graphql/index.ts

```diff
@@ -4,15 +4,56 @@ import { UserModel } from '../models/UserProfile';
 
 const UserTC = composeWithMongoose(UserModel, {});
 
+function requireToken(resolvers) {
+  return Object.keys(resolvers).reduce((acc, key) => {
+    return {
+      ...acc,
+      [key]: resolvers[key].wrapResolve(next => rp => {
+        if (!rp.context.jwt.valid) {
+          throw new Error('You must provide valid token');
+        }
+        return next(rp);
+      }),
+    };
+  }, {});
+}
+
+function requireSameUserMongoId(resolvers) {
+  return Object.keys(resolvers).reduce((acc, key) => {
+    return {
+      ...acc,
+      [key]: resolvers[key].wrapResolve(next => async rp => {
```

Comment:
what is `rp`?

Reply:
hmm, I copied it from graphqlCompose, my guess is resolverParameters. I'll destructure it so I can avoid the naming :D